source/ai/rl/BBF_agent/BBF.py

```
@@ -0,0 +1,442 @@
+                    ret = 0
+                    for u in reversed(range(t, n)):
+                        # print(c_flag.shape, gammas.shape, rewards.shape)
+                        ret += (
+                            torch.prod(c_flag[:, t + 1 : u + 1], -2)
+                            * torch.prod(gammas[:, t:u], -2)
+                            * rewards[:, u + 1]
+                        )
+                    returns.append(ret)
+                returns = torch.stack(returns, 1)
+
+            plot_vs = returns.clone().sum(-1)
```

Ruff (0.12.2)

Local variable `plot_vs` is assigned to but never used

Remove assignment to unused variable `plot_vs`

(F841)